packages/web/components/bridge/immersive/asset-select-screen.tsx

```diff
@@ -0,0 +1,226 @@
+  ({ type, onSelectAsset }: AssetSelectScreenProps) => {
+    const { accountStore, userSettings } = useStore();
+    const { showPreviewAssets } = useShowPreviewAssets();
+    const { t } = useTranslation();
+
+    const wallet = accountStore.getWallet(accountStore.osmosisChainId);
+
+    const [search, setSearch] = useState("");
+    const [assetToActivate, setAssetToActivate] = useState<MinimalAsset | null>(
+      null
+    );
+
+    const showUnverifiedAssetsSetting =
+      userSettings.getUserSettingById<UnverifiedAssetsState>(
+        "unverified-assets"
+      );
+    const shouldShowUnverifiedAssets =
+      showUnverifiedAssetsSetting?.state.showUnverifiedAssets;
+
+    const { data, isLoading, fetchNextPage, hasNextPage, isFetchingNextPage } =
+      api.edge.assets.getImmersiveBridgeAssets.useInfiniteQuery(
+        {
+          type,
+          search: Boolean(search)
+            ? {
+                query: search,
+              }
+            : undefined,
+          userOsmoAddress: wallet?.address,
+          includePreview: showPreviewAssets,
+          variantsNotToBeExcluded,
+          prioritizedDenoms,
+          limit: 50, // items per page
+        },
+        {
+          getNextPageParam: (lastPage) => lastPage.nextCursor,
+          initialCursor: 0,
+
+          // avoid blocking
+          trpc: {
+            context: {
+              skipBatch: true,
+            },
+          },
+        }
+      );
+
+    const assets = useMemo(
+      () => data?.pages.flatMap((page) => page?.items) ?? [],
+      [data?.pages]
+    );
+    const canLoadMore = !isLoading && !isFetchingNextPage && hasNextPage;
+
+    return (
+      <div>
+        <ActivateUnverifiedTokenConfirmation
+          coinDenom={assetToActivate?.coinDenom}
+          coinImageUrl={assetToActivate?.coinImageUrl}
+          isOpen={Boolean(assetToActivate)}
+          onConfirm={() => {
+            if (!assetToActivate) return;
+            showUnverifiedAssetsSetting?.setState({
+              showUnverifiedAssets: true,
+            });
+            onSelectAsset(assetToActivate);
+          }}
+          onRequestClose={() => {
+            setAssetToActivate(null);
+          }}
+        />
+
+        <h1 className="text-center text-h5 font-h5">
+          {t(
+            type === "deposit"
+              ? "transfer.assetSelectScreen.titleDeposit"
+              : "transfer.assetSelectScreen.titleWithdraw"
+          )}
+        </h1>
+
+        <SearchBox
+          onInput={debounce((nextValue) => {
+            setSearch(nextValue);
+          }, 300)}
+          className="my-4 flex-shrink-0"
+          placeholder={t("transfer.assetSelectScreen.searchAssets")}
+          size="full"
+        />
+
+        <div className="flex flex-col gap-1">
+          {isLoading ? (
+            <>
+              {new Array(7).fill(undefined).map((_, i) => (
+                <SkeletonLoader key={i} className="h-[80px] w-full" />
```

Comment:
Is this what was used in the design? IMO the feel of the skeleton loader feels a bit jarring

Reply:
Yes, this can be improved. My plan is to finish the flow and then do a design check with the team for some final refinements